lib/fluent/plugin/kinesis_helper/client.rb

```diff
@@ -139,6 +139,8 @@ def setup_credentials
                 credentials_options[:client] = Aws::STS::Client.new(region: @region, http_proxy: c.sts_http_proxy)
             elsif @region and c.sts_endpoint_url
                 credentials_options[:client] = Aws::STS::Client.new(region: @region, endpoint: c.sts_endpoint_url)
+            elsif @region and c.sts_http_proxy and c.sts_endpoint_url
```

Comment:
This condition (`elsif @region and c.sts_http_proxy and c.sts_endpoint_url`) will not work because compatible condition in L138 (`elsif @region and c.sts_http_proxy`) is defined before this line. Can you confirm that?

Reply:
Thank you for your feedback, :)